### PR TITLE
fix-editor-comments: add comment snapping and leashing

### PR DESCRIPTION
**Pull request by lisa-wolfgang**
Resolves https://github.com/ScratchAddons/ScratchAddons/issues/3782

### Changes

Two new features have been added to `fix-editor-comments`: "Always spawn block comments next to block" (leashing) and "Keep block comment connections straight" (snapping).

**Before leashing:**

https://user-images.githubusercontent.com/43426138/141669491-2ce2bdcc-2f8f-496b-afa6-27c2c3815033.mp4

**After leashing:**

https://user-images.githubusercontent.com/43426138/141669490-028debc8-349e-4620-b5b1-2fb728121d6e.mp4

Here's what snapping looks like:

https://user-images.githubusercontent.com/43426138/141669488-03ca0f95-0e54-4c11-9551-9a537f374895.mp4

### Reason for changes

Off-screen comments are extremely annoying. They are one of the deterrents that keep people from using block comments.
Snapping also makes comments a lot neater-looking while still allowing the user to move them horizontally, which also makes block comments more appealing to use.

### Tests

Tested on browser w/ Chromium 95 and Firefox 94.0.1.

### CODE REVIEW

File: addons/fix-editor-comments/addon.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     },
     {
       "name": "lisa_wolfgang",
-      "link": "https://github.com/lisa-wolfgang"
+      "link": "https://scratch.mit.edu/lisa_wolfgang"
```
Comment (Samq64) [inline addons/fix-editor-comments/addon.json:12]:
```suggestion
      "link": "https://scratch.mit.edu/users/lisa_wolfgang/"
```

**Comment by lisa-wolfgang** [comment]
TODO: When https://github.com/ScratchAddons/ScratchAddons/pull/3811 is merged, add `latestUpdate` property to manifest.

File: addons/fix-editor-comments/addon.json [hunk 2]
```diff
@@ -52,6 +52,18 @@
       "id": "standalone-edit",
       "type": "boolean",
       "default": true
+    },
+    {
+      "name": "Always spawn block comments next to block",
+      "id": "leash",
+      "type": "boolean",
+      "default": true
+    },
```
Comment (WorldLanguages) [inline addons/fix-editor-comments/addon.json:61]: Are we sure all old users of this addon will be fine with this being enabled automatically?

Comment (WorldLanguages) [reply addons/fix-editor-comments/addon.json:61]: Leashing does not happen that often, and it's easy to trace the behavior to this addon, so I'd say it's fine.

Comment (lisa-wolfgang) [reply addons/fix-editor-comments/addon.json:61]: Very few people would want a new comment to be created off-screen automatically. There is little purpose to it, and it is extremely annoying in most cases.

**Pull request merged**